Source/Spatializer/HoloLens2/IsacAdapter.cpp

```diff
@@ -287,6 +304,8 @@ HRESULT UpdateSpatialAudioObject(IsacSpatialSourceInternal* source)
 // Remove any sources that have been marked for deletion
 void IsacAdapter::PruneStaleSources()
 {
+    std::scoped_lock lock(m_DeviceChangeLock);

```

Comment:
PruneStaleSources() gets called on the ISAC pump thread. Are you worried at all about taking a lock here?

Reply:
What was the behavior in the Wwise plugin? It's the exact same sequence so if it didn't cause issues there I won't be worried. Also this is narrowly taken on stream switch so maybe it's ok?

Reply:
Ah, didn't realize this was just a port from there. Didn't notice any issues in Wwise from this. Probably fine.